pkg/pluginmanager/manager.go

```diff
@@ -989,7 +996,50 @@ func SyncPlugins() error {
 	if err != nil {
 		errList = append(errList, err)
 	}
-	if installedPlugins, err := pluginsupplier.GetInstalledServerPlugins(); err == nil {
+	err = installDiscoveredContextPlugins(plugins)
+	if err != nil {
+		errList = append(errList, err)
+	}
+	err = kerrors.NewAggregate(errList)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// SyncPluginsForTarget installs the plugins for given target
+func SyncPluginsForTarget(target configtypes.Target) error {
+	currentContextMap, err := configlib.GetAllCurrentContextsMap()
+	if err != nil {
+		return err
+	}
+	ctx, ok := currentContextMap[target]
+	if !ok {
+		return fmt.Errorf("there is no active context for the target %v ", target)
+	}
+	log.Infof("Checking for required plugins for context %s...", ctx.Name)
+	errList := make([]error, 0)
+	plugins, err := DiscoverServerPluginsForGivenContexts([]*configtypes.Context{ctx})
+	if err != nil {
+		errList = append(errList, err)
+	}
+	err = installDiscoveredContextPlugins(plugins)
+	if err != nil {
+		errList = append(errList, err)
+	}
+	err = kerrors.NewAggregate(errList)
```

Comment:
I believe you can simply do `return  kerrors.NewAggregate(errList)`

Reply:
updated